app/services/duty_options/base.rb

```diff
@@ -63,7 +63,14 @@ def measure_unit?
     end
 
     def total_quantity
-      number_with_precision(duty_evaluation[:total_quantity], precision: 10, significant: false, strip_insignificant_zeros: true)
+      if /[\.e]/ !~ duty_evaluation[:total_quantity].to_s || /\.\d$/ =~ duty_evaluation[:total_quantity].to_s
```

Comment:
How do you feel about moving this to a reusable service?

```ruby

class MeasureComponentQuantityFormatService
  include ActionView::Helpers::NumberHelper

  def initialize(quantity)
    @quantity = quantity
  end

  def call
    number_with_precision(
      @quantity,
      precision: precision,
      significant: false,
      strip_insignificant_zeros: strip_zeroes,
    )
  end

  private

  def precision
    return 10 if scientific_notation?

    2
  end

  def strip_insignificant_zeros
    return true if scientific_notation?

    false
  end

  def scientific_notation?
    @scientific_notation ||= /[.e]/ =~ quantity.to_s || /\.\d$/ !~ quantity.to_s
  end
end
```

We can then write unit tests more easily and just consume it in the duty options and the evaluator?

Reply:
Sounds like a good idea, but I would keep the name more generic - it may end up being useful for other quantities other than the measure amount.
How does `NumberWithHighPrecisionFormatter` sound?

Reply:
Sounds perfect :)